bdns/dns.go

```diff
@@ -277,48 +277,60 @@ func (dnsClient *impl) exchangeOne(ctx context.Context, hostname string, qtype u
 	chosenServerIndex := 0
 	chosenServer := servers[chosenServerIndex]
 
+	// Strip off the IP address part of the server address because
+	// we talk to the same server on multiple ports, and don't want
+	// to blow up the cardinality.
+	chosenServerIP, _, err := net.SplitHostPort(chosenServer)
+	if err != nil {
+		return
+	}
+
 	start := dnsClient.clk.Now()
 	client := dnsClient.dnsClient
 	qtypeStr := dns.TypeToString[qtype]
 	tries := 1
 	defer func() {
-		result, authenticated := "failed", ""
+		result := "failed"
 		if resp != nil {
 			result = dns.RcodeToString[resp.Rcode]
-			authenticated = fmt.Sprintf("%t", resp.AuthenticatedData)
 		}
 		dnsClient.totalLookupTime.With(prometheus.Labels{
-			"qtype":              qtypeStr,
-			"result":             result,
-			"authenticated_data": authenticated,
-			"retries":            strconv.Itoa(tries),
-			"resolver":           chosenServer,
+			"qtype":    qtypeStr,
+			"result":   result,
+			"retries":  strconv.Itoa(tries),
+			"resolver": chosenServerIP,
 		}).Observe(dnsClient.clk.Since(start).Seconds())
 	}()
 	for {
 		ch := make(chan dnsResp, 1)
 
+		// Strip off the IP address part of the server address because
+		// we talk to the same server on multiple ports, and don't want
+		// to blow up the cardinality.
+		chosenServerIP, _, err = net.SplitHostPort(chosenServer)
+		if err != nil {
+			return
+		}
```

Comment:
Please include a note which indicates that this return should never happen because `validateServerAddress()` ensures that a server address can never be implicit localhost (e.g. `:1337`)

Reply:
Without this note I was concerned that we could end up with: https://go.dev/play/p/MHcSK5Dh0eP